src/main/java/nl/_42/max_login_attempts_spring_boot_starter/service/LoginAttemptServiceImplementation.java

```diff
@@ -98,17 +93,14 @@ private boolean hasReachedLoginAttemptLimit(String username, String remoteAddres
      */
     @Override
     public void loginSucceeded(String username, String remoteAddress) {
-        LOGGER.debug("User {} on remote address {} succeeded to login.", username, remoteAddress);
+        log.debug("User {} on remote address {} succeeded to login.", username, remoteAddress);
         UsernameIPAddress usernameIPAddress = new UsernameIPAddress(username, remoteAddress);
-        clearAttempts(usernameIPAddress);
-    }
-
-    private void clearAttempts(UsernameIPAddress usernameIPAddress) {
-        attemptsCache.remove(usernameIPAddress);
+        resetAttempts(usernameIPAddress);
     }
 
-    private void clearBlocked(UsernameIPAddress usernameIPAddress) {
-        blockedUsernameIPAddresses.remove(usernameIPAddress);
+    private void resetAttempts(UsernameIPAddress usernameIPAddress) {
+        Attempt attempt = userCache.getOrDefault(usernameIPAddress, new Attempt());
```

Comment:
Is het nodig om deze opnieuw uit de cache te halen? Dat doe je in regel 62 toch al?

Reply:
deze wordt gebruikt door loginFailed(), loginSucceeded() en resetByUsername() en deze maken (op de loginFailed na) de UsernameIPAddress aan zonder het eerst uit de userCache te halen

Reply:
Dan zou ik als argument een AttemptsMonitor meegeven. Die in die andere gevallen dus nog uit de cache moet worden gehaald.